### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by brenthuisman**
- Update Arbor to v0.8
  - Update for some minor API changes therein.
- Simplify and fix CI
  - PR workflow now also builds AppImage/DiskIm, since it's a small overhead
  - No more Nightly because of that
  - Tell CMake to `-DBUILD_SHARED_LIBS=OFF`, to cope with a submodule (glm) incorrectly globally setting this, causing lib packaging errors when building AppImage/DiskImages.
  - Break out the AppImage script into workflow steps, to help troubleshoot breakage.
  - Update minimum build-OSes; GitHub have deprecated the macos-10.15 and ubuntu-18.04 runners.

### CODE REVIEW

File: .github/workflows/artifact-release.yml [hunk 1]
```diff
@@ -6,48 +6,92 @@ on:
...
-    runs-on: ubuntu-18.04
+    name: "Build on Ubuntu"
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config:
+        - { 
+            name: "Ubuntu GCC",
+            os:   "ubuntu-20.04",
+            cc:   "gcc-10",
```
Comment (thorstenhater) [inline .github/workflows/artifact-release.yml:21]: This choice seems rather conservative, if there's no particular reason, maybe go more modern?

File: .github/workflows/artifact-release.yml [hunk 2]
```diff
@@ -6,48 +6,92 @@ on:
...
-    name: "Make Apple Disk Image Nightly"
-    runs-on: macos-10.15
+    name: "Build on MacOS"
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config:
+        - {
+            name: "MacOS",
+            os:   "macos-11",
```
Comment (thorstenhater) [inline .github/workflows/artifact-release.yml:70]: Similarly here: I am personally on MacOS 12 and 13 has been released.
Do we support these version when build on 11?

**Review by thorstenhater — CHANGES_REQUESTED** [review]
Some questions on the versions, rest looks great. Thanks a lot!

Question about the GLFW hiccup. Is setting `BUILD_SHARED_LIBS` before
`add_subdirectory(glfw)` enough to fix it?

File: .github/workflows/artifact-release.yml (hunk 1 above)
Comment (brenthuisman) [reply .github/workflows/artifact-release.yml:21]: The Linuxdeploy script target a minimum GLIB based on the OS of building. Therefore, the resulting AppImage is compatible with GLIBs as old as your build OS and up. 20.04 is already quite modern (CentOS users are probably left out in the cold, which is potentially a problem, it's also the desktop OS in many an institute that rools out Linux machines (Scientific Linux, CernOS etc).

So, it's too modern, if anything :P Maybe the clang build could be bumped, use that job for testing of updated tooling?

File: .github/workflows/artifact-release.yml (hunk 2 above)
Comment (brenthuisman) [reply .github/workflows/artifact-release.yml:70]: macos-11 and -12 are the only Runners Github provides (and 10.15, which will disappear in a few months).

The Disk Image should however run on more recent OSes (I suppose something similar to GLIB applies, OS often keeps API/ABI-compat with older releases, at least for some time), but I can't confirm that due to lack of mac. I can confirm that for Simon (who runs 11) a DiskIm built on 12 did not work (and built on 11 does).

**Comment by brenthuisman** [comment]
On `BUILD_SHARED_LIBS` before `add_subdirectory(glfw)`: you mean instead of passing the flag to CMake?

The lib messing stuff up is `glm`, not `glfw`. But I can't find a `add_subdirectory(glm)` anywhere in the project, only `target_include_directories(arbor-gui-deps SYSTEM INTERFACE 3rd-party/glm)` (header only?). Would setting it before this line be what you want to try?

File: .github/workflows/artifact-release.yml (hunk 1 above)
Comment (thorstenhater) [reply .github/workflows/artifact-release.yml:21]: Ok, I'll take that explanation. This project needs C++20, though, so there is a sweet spot somewhere in the middle.

File: .github/workflows/artifact-release.yml (hunk 2 above)
Comment (thorstenhater) [reply .github/workflows/artifact-release.yml:70]: Yes, Mac compilers usually have this toggle 'Minimum Deploy Target' specifying the minimal version of MacOS the artifact should be compatible with.

**Comment by thorstenhater** [comment]
Yes `glm` is header only, so it cannot set `BUILD_SHARED_LIBS=ON`, because we never executed their CMake. Thus, `glm` cannot be the issue.

_Also_ I use `glm` since the beginning of this project. `glbinding` is new though, so I start poking there and they seem
to be the culprit
```
3rd-party/glbinding/CMakeLists.txt
86:option(BUILD_SHARED_LIBS              "Build shared instead of static libraries."                 ON)
```

**Comment by brenthuisman** [comment]
OK, let's see if that works.

**Comment by brenthuisman** [comment]
So, jobs are passing, AppImage #worksforme, but the DiskIm does not work on @thorstenhater's M1.

**Review by thorstenhater — APPROVED** [review]
I'll accept, but we need to fix the M1 issue(s). These are getting more and more common.

**Pull request merged**